javascript/ecmascript-2015/promises/introducing-promises.md

```diff
@@ -24,25 +24,53 @@ links:
 ---
 ## Content
 
-A **promise** is an object that represents a value that may or may not be obtained in the future and can be passed around like any other object.
+A **promise** is an object that represents a value that may or may not be obtained in the future and can be passed around like any other value.
 
-You may have come across promises in JavaScript before either in the language itself, libraries such as jQuery (which some would argue are not really promises) and libraries such as q or rsvp.
+### Why use promises?
 
-Some of these libraries conform (and add their own additional functionality as well) to a specification called **Promise/A+**. ES6 bases its promise implementation on this specification.
+It is common in JavaScript to perform tasks that do not complete immediately – for example making an ajax call. Pages would run very slowly if we had to block everything to wait for an ajax call to finish.
 
-**Why use promises?**
+However we still need a way to know when our ajax call has completed. One way of approaching this problem is to use callback functions. A callback function is called when the original task has completed.
 
-It is common in JavaScript to perform tasks that do not complete immediately – for example making an ajax call.
+The most common approach in JavaScript to perform an async operation is to use a callback with an async mechanism like fetch or setTimeout. Callbacks work great for simple scenarios such as doing one or two async tasks at a time. As soon as we need to involve more async tasks or increase their complexity with logic and/or error handling, callbacks-based code starts getting difficult to manage.
 
-JavaScript only has one thread (ignoring web and service workers) to run in. Pages would run very slowly if we had to stop everything on a page to wait for an ajax call so JavaScript continues to execute other code.
+With Promises we do not get this issue because promises are inherently reliable (they handle errors for us automatically) and they allow us to write simpler syntax for handling async logic
 
-However we still need a way to know when our ajax call has completed.
+At the same time, Promises we can write code that pretends like we already have the value, no matter when that value actually arrives. This allows Promises to eliminate time as a concern in our code.
```

Comment:
add "with" before Promises, use "behave" instead of "pretend":

> At the same time, with Promises we can write code that behaves...

Reply:
done